src/main.cpp

```diff
@@ -57,47 +57,47 @@ static struct options opt = {
-
-static void help(FILE *fp) {
-    fprintf(fp,
-            "%s [OPTIONS...] [<uart>|<udp_address>]\n\n"
-            "  <uart>                       UART device (<device>[:<baudrate>]) that will be routed\n"
-            "  <udp_address>                UDP address (<ip>:<port>) that will be routed\n"
-            "  -e --endpoint <ip[:port]>    Add UDP endpoint to communicate port is optional\n"
-            "                               and in case it's not given it starts in 14550 and\n"
-            "                               continues increasing not to collide with previous\n"
-            "                               ports\n"
-            "  -p --tcp-endpoint <ip:port>  Add TCP endpoint client, which will connect to given\n"
-            "                               address\n"
-            "  -r --report_msg_statistics   Report message statistics\n"
-            "  -t --tcp-port <port>         Port in which mavlink-router will listen for TCP\n"
-            "                               connections. Pass 0 to disable TCP listening.\n"
-            "                               Default port 5760\n"
-            "  -c --conf-file <file>        .conf file with configurations for mavlink-router.\n"
-            "  -d --conf-dir <dir>          Directory where to look for .conf files overriding\n"
-            "                               default conf file.\n"
-            "  -l --log <directory>         Enable Flight Stack logging\n"
-            "  -g --debug-log-level <level> Set debug log level. Levels are\n"
-            "                               <error|warning|info|debug>\n"
-            "  -v --verbose                 Verbose. Same as --debug-log-level=debug\n"
-            "  -V --version                 Show version\n"
-            "  -h --help                    Print this message\n"
-            , program_invocation_short_name);
+static const struct option long_options[] = {{"endpoints", required_argument, NULL, 'e'},
+                                             {"conf-file", required_argument, NULL, 'c'},
+                                             {"conf-dir", required_argument, NULL, 'd'},
+                                             {"report_msg_statistics", no_argument, NULL, 'r'},
+                                             {"tcp-port", required_argument, NULL, 't'},
+                                             {"tcp-endpoint", required_argument, NULL, 'p'},
+                                             {"log", required_argument, NULL, 'l'},
+                                             {"debug-log-level", required_argument, NULL, 'g'},
+                                             {"verbose", no_argument, NULL, 'v'},
+                                             {"version", no_argument, NULL, 'V'},
+                                             {}};
+
+static const char *short_options = "he:rt:c:d:l:p:g:vV";
+
+static void help(FILE *fp)
+{
+    fprintf(
+        fp,
```

Comment:
don't really understand the logic of moving fp to the next line... what config is causing this?